hydroflow_lang/src/pretty_span.rs

```diff
@@ -3,22 +3,36 @@
 pub struct PrettySpan(pub proc_macro2::Span);
 impl std::fmt::Display for PrettySpan {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        let span = self.0.unwrap();
-        write!(
-            f,
-            "{}:{}:{}",
-            span.source_file().path().display(),
-            span.start().line,
-            span.start().column
-        )
+        let span = self.0;
+        #[cfg(feature = "diagnostics")]
+        {
+            write!(
+                f,
+                "{}:{}:{}",
+                span.unwrap().source_file().path().display(),
+                span.start().line,
+                span.start().column
+            )
+        }
+
+        #[cfg(not(feature = "diagnostics"))]
+        {
+            write!(
+                f,
+                "{}:{}:{}",
+                "unknown",
+                span.start().line,
+                span.start().column
+            )
+        }
```

Comment:
Hmm would prefer to keep one `write!` and gate inside, or even just `.unwrap_or_else(|| "unknown".to_owned())`
```rust
write!(
    f,
    "{}:{}:{}",
    { // not sure about the exact hygene of these but something like this
        #[cfg(feature = "diagnostics")]
        span.unwrap().source_file().path().display()
        #[cfg(not(feature = "diagnostics"))]
        "unknown"
    },
    span.start().line,
    span.start().column
)
```